avocado/core/mux.py

```diff
@@ -163,15 +164,25 @@ def initialize_mux(self, root, mux_path, debug):
         self.root = root
         self.mux_path = mux_path
         self.debug = debug
+        self.variant_ids = self._get_variant_ids()
+
+    def _get_variant_ids(self):
+        variant_ids = []
+        for variant in MuxTree(self.root):
+            variant.sort(key=lambda x: x.path)
+            fingerprint = "-".join(_.fingerprint() for _ in variant)
+            variant_ids.append("-".join(node.name for node in variant) + '-' +
+                               sha.sha(fingerprint).hexdigest()[:4])
```

Comment:
What's the intention behind this hash? Isn't the variant id already unique? It's kinda easy to have clashes in a 4 digits trunked sha1 string, so I don't see the point in having that. For instance, this yaml:
```yaml 
l1: !mux
    v1:
        var: 1
l2: !mux
    v1:
        var: 2048
    v2:
        var: 3519
```

will produce these variants:

```
Multiplex variants:
Variant v1-v1-0c0a:    /run/l1/v1, /run/l2/v1
Variant v1-v2-0c0a:    /run/l1/v1, /run/l2/v2
```